src/test/java/smokefree/domain/RecordPlaygroundObservationCommandTest.java

```diff
@@ -0,0 +1,60 @@
+package smokefree.domain;
+
+
+
+import io.micronaut.test.annotation.MicronautTest;
+import org.junit.jupiter.api.Test;
+import smokefree.aws.rds.secretmanager.SmokefreeConstants;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.ValidatorFactory;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@MicronautTest
+public class RecordPlaygroundObservationCommandTest {
+
+	private ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
+
+	@Test
+    public void should_allow_when_commentLengthIsBetween1To2000() {
+		RecordPlaygroundObservationCommand recordPlaygroundObservationCommand = new RecordPlaygroundObservationCommand("123","dev-user",true, "Smokefree playground");
+		Set<ConstraintViolation<RecordPlaygroundObservationCommand>> violations =  validatorFactory.getValidator().validate(recordPlaygroundObservationCommand);
+		assertEquals(0, violations.size());
+	}
+
+	@Test
+	public void should_notAllow_when_messageContainsNotAllowedCharacters() {
+		RecordPlaygroundObservationCommand recordPlaygroundObservationCommand = new RecordPlaygroundObservationCommand("123","dev-user",true, "~");
+		Set<ConstraintViolation<RecordPlaygroundObservationCommand>> violations = validatorFactory.getValidator().validate(recordPlaygroundObservationCommand);
+		assertEquals("Please enter only allowed special charaxters: @&(),.?\": ", violations.iterator().next().getMessage());
+		assertEquals(1, violations.size());
+
+	}
+
+	@Test
+    public void should_notAllow_when_commentIsBlank() {
+		RecordPlaygroundObservationCommand recordPlaygroundObservationCommand = new RecordPlaygroundObservationCommand("123","dev-user",true, "    ");
```

Comment:
drop

Reply:
I have removed it in the other PR and it will be fine when I rebase the branch